tools/cldr-code/src/main/java/org/unicode/cldr/util/Containment.java

```diff
@@ -214,11 +214,19 @@ private static void getAllDirected(Multimap<String, String> multimap, String lan
             targets.add(target);
         } else if (size == 1) {
             for (String parent : parents) {
-                getAllDirected(multimap, parent, target, targets);
+                if (parent.equals(lang)) {
+                    System.err.println("ERR: "+ lang + " is its own parent");
+                } else {
+                    getAllDirected(multimap, parent, target, targets);
+                }
             }
         } else {
             for (String parent : parents) {
-                getAllDirected(multimap, parent, (ArrayList<String>) target.clone(), targets);
+                if (parent.equals(lang)) {
+                    System.err.println("ERR: "+ lang + " is its own parent");
+                } else {
+                    getAllDirected(multimap, parent, (ArrayList<String>) target.clone(), targets);
+                }
```

Comment:
https://www.wikidata.org/wiki/Q2873230 is a subcategory of itself

Reply:
I assume this is a bug, but asked https://www.wikidata.org/wiki/Property_talk:P279#P279_is_not_reflexive

Reply:
The wikidata often contains cycles (they have no test that ensure a clean tree).

Reply:
Looks like an excellent speedup.